File: c/Lesson_6_TiltSwitch/6_Tilt.c
/**********************************************************************
* Filename    : 6_VibrationTilt.c
* Description : Use Tilt to control LED.
* Author      : Robot
* E-mail      : support@sunfounder.com
* website     : www.sunfounder.com
* Update      : Cavon    2016/07/01
**********************************************************************/

#include <wiringPi.h>
#include <stdio.h>

#define TiltPin		0
#define Gpin		2
#define Rpin		3

int main(void)
{
	

	printf("\n");
	printf("\n");
	printf("========================================\n");
	printf("|              Tilt Switch             |\n");
	printf("|    ------------------------------    |\n");
	printf("|      Tilt connect to GPIO 0(pin 11)  |\n");
	printf("|   Green LED connect to GPIO 2(pin 13)|\n");
	printf("|    Red LED connect to GPIO 3(pin 15) |\n");
	printf("|                                      |\n");
	printf("|        Use Tilt to control LED       |\n");
	printf("|                                      |\n");
	printf("|                            SunFounder|\n");
	printf("========================================\n");
	printf("\n");
	printf("\n");
	
	if(wiringPiSetup() == -1){ //when initialize wiring failed,print messageto screen
		printf("setup wiringPi failed !");
		return 1; 
	}

	pinMode(TiltPin, INPUT);
	pinMode(Gpin, OUTPUT,);
	pinMode(Rpin, OUTPUT);


	while(1){
		
		if(1 == digitalRead(TiltPin)){
			delay(10);
			if(1 == digitalRead(TiltPin)){
				digitalWrite(Rpin, HIGH);
				digitalWrite(Gpin, LOW);
				printf("RED\n");
			}
		}
		else if(0 == digitalRead(TiltPin)){
			delay(10);
			if(0 == digitalRead(TiltPin)){
				while(!digitalRead(TiltPin));				
				digitalWrite(Rpin, LOW);
				digitalWrite(Gpin, HIGH);
				printf("GREEN\n");
			}
		}
	}
	return 0;
}

File: for-raspberry-pi/python/6_Tilt.py
#!/usr/bin/env python3
import RPi.GPIO as GPIO

TiltPin = 17
Gpin   = 27
Rpin   = 22

def print_message():
	print ("========================================")
	print ("|           Tilt control LED           |")
	print ("|    ------------------------------    |")
	print ("|         Tilt connect to BCM 17       |")
	print ("|      Green LED connect to BCM 27     |")
	print ("|        Red LED connect to BCM 22     |")
	print ("|                                      |")
	print ("|       User Tilt to control LED       |")
	print ("|                                      |")
	print ("|                            SunFounder|")
	print ("========================================\n")
	print ('Program is running...')
	print ('Please press Ctrl+C to end the program...')
	input ("Press Enter to begin\n")

def setup():
	GPIO.setmode(GPIO.BCM)       # Numbers GPIOs by BCM
	GPIO.setup(Gpin, GPIO.OUT,initial=GPIO.HIGH)     # Set Green Led Pin mode to output
	GPIO.setup(Rpin, GPIO.OUT,initial=GPIO.LOW)     # Set Red Led Pin mode to output
	GPIO.setup(TiltPin, GPIO.IN)    # Set BtnPin's mode is input
	GPIO.add_event_detect(TiltPin, GPIO.BOTH, callback=detect, bouncetime=200)

def Led(x):
	if x == 0:
		GPIO.output(Rpin, 1)
		GPIO.output(Gpin, 0)
	if x == 1:
		GPIO.output(Rpin, 0)
		GPIO.output(Gpin, 1)

def Print(x):
	if x == 0:
		print ('    *************')
		print ('    *   Tilt!   *')
		print ('    *************')

def detect(chn):
	Led(GPIO.input(TiltPin))
	Print(GPIO.input(TiltPin))

def loop():
	while True:
		pass

def destroy():
	GPIO.output(Gpin, GPIO.LOW)       # Green led off
	GPIO.output(Rpin, GPIO.LOW)       # Red led off
	GPIO.cleanup()                     # Release resource

if __name__ == '__main__':     # Program start from here
	print_message()
	setup()
	try:
		loop()
	except KeyboardInterrupt:  # When 'Ctrl+C' is pressed, the child program destroy() will be  executed.
		destroy()

